File: Invoices/Models/WorkItem.cs
using System;
using System.Collections.Generic;

namespace Invoices.Models
{
   public class WorkItem
    {
        public virtual int Id { get; }  
        public virtual int? ParentId { get; set; }
        public virtual int ExternalId { get; init; }
        public virtual DateTime CreatedDate { get; init; }    
        public virtual DateTime? LastUpdateTime { get; set; }
        public virtual DateTime? ClosedDate { get; set; }
        public virtual string Title { get; init; }
        public virtual string Type { get; init; }
        public virtual string Status { get; set; }
        public virtual ICollection<HistoryDetail> History { get; set; }

    }
}

File: Invoices/ClientApp/src/app/models/WorkItem.ts
import { HistoryDetail } from "./HistoryDetail";

export class WorkItem {
    constructor(
        public externalId?:number,
        public name?: string,
        public createdDate?:Date,
        public lastUpdatedDate?:Date,
        public closedDate?:Date,
        public title?:string,
        public type?:string,
        public status?:string,
        public History?: HistoryDetail[]) { }
}